recipes/zmqpp/all/conanfile.py

```diff
@@ -0,0 +1,74 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class ZmqppConan(ConanFile):
+    name = "zmqpp"
+    homepage = "https://github.com/zeromq/zmqpp"
+    license = "MPL-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "This C++ binding for 0mq/zmq is a 'high-level' library that hides most of the c-style interface core 0mq provides."
+    topics = ("conan", "zmq", "0mq", "zeromq", "message-queue", "asynchronous")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    exports_sources = "CMakeLists.txt", "patches/**"
+    generators = ["cmake", "cmake_find_package"]
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("libsodium/1.0.18")
+        self.requires("zeromq/4.3.3")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("zmqpp-%s" % (self.version), self._source_subfolder)
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def validate(self):
+        compiler = self.settings.compiler
+        if compiler.get_safe('cppstd'):
+            tools.check_min_cppstd(self, 11)
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("Visual Studio compiler is not supported")
```

Comment:
Are you sure? zeromq explicitly doesn't support Visual Studio, or the build is just broken for some triplets?

Reply:
The build is broken for MSVS. After some hacks applied to build configuration, it starts compilation step, but compilation errors appear. So I disabled MSVS support for now.

Reply:
I think that zmqpp 4.2.0 may support some (old?) versions of MSVS, but it would still require some tweaks/patches in Conan recipe to support that.